### PR TITLE
Add sitemaps

### PR DESCRIPTION
**Pull request by piotrjakimiak**
Add sitemaps (Fix issue #175)

### CODE REVIEW

File: aldryn_newsblog/models.py [hunk 1]
```diff
@@ -183,7 +183,7 @@ def get_absolute_url(self, language=None):
         # articles before the page exists.
         #
         if language is None:
-            language = get_language()
+            language = get_current_language()
```
Comment (piotrjakimiak) [inline aldryn_newsblog/models.py]: It breaks tests, so i fixed it.

Comment (mkoistinen) [inline aldryn_newsblog/models.py]: Looks like several of us made the same fix at the same time =)

**Comment by mkoistinen** [comment]
Does this support multiple languages? Also, does this support multiple app-hooks?

File: aldryn_newsblog/sitemaps/sitemap.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+from ..models import Article
+
+
+class NewsBlogSitemap(Sitemap):
+
+    changefreq = "never"
+    priority = 0.5
+
+    def __init__(self, *args, **kwargs):
+        self.namespace = kwargs.pop('namespace', None)
```
Comment (mkoistinen) [inline aldryn_newsblog/sitemaps/sitemap.py]: Please add `self.language = kwargs.pop('language', None)` here.

**Comment by piotrjakimiak** [comment]
@mkoistinen I've update PR, with support for languages and added test for it.

**Comment by mkoistinen** [comment]
Awesome!

**Comment by mkoistinen** [comment]
I notice that you haven't overridden location(). I suspect this won't work correctly without that. I think we'd like the search engines to list the articles in all the languages they're available in, not just one language, right?

**Comment by piotrjakimiak** [comment]
I tested using override here but it doesn't work as I expect. When language is set it returns url for given language if it exists or fallback otherwise. 
For example sitemap for 'de' language in my test after changes returns:
[u'http://example.com/de/page/vjqDbJtyntHlTfATinXwHRA/', u'http://example.com/en/page/cuZsBXGQzUgExANsZdUPtcW/', u'http://example.com/de/page/de-article/']
(Second element has no de version)

I think what we should get is:
* if language for sitemap is set return all articles in this language version
* otherwise return url for all language versions of document

Am I right or You prefere other solutions?

**Comment by mkoistinen** [comment]
In the example code I posted for what to put in urls.py, it shows how the sitemaps will be created for each language. So, location() should never really be called without a language set.

**Comment by piotrjakimiak** [comment]
Ok I will do it that way.

But what should be the correct result for that situation:
Site has two languages 'en' and 'de'

A1 has translations for 'en' and 'de'
A2 has translations for 'en'
A3 has translations for 'de'

What should be results for 
`NewsBlogSitemap(language='en')`?
A1 and A2 (urls for en versions)?

`NewsBlogSitemap(language='en')`?
A1 and A3 (urls for de versions)?

**Comment by mkoistinen** [comment]
I think that would be correct. And, we should probably do this regardless of the fallback preferences set in CMS_LANGUAGES. Google, et al, shouldn't index pages that are just fallbacks for a missing language. If nothing else, this would be bad for SEO.

**Comment by piotrjakimiak** [comment]
@mkoistinen I've updated the PR.

I'm using `translated()` method to get only articles translated to given language.
In location language is passed into `get_absolute_url()` to make sure correct language is returned (tests check it).
It now works as I described in my previous comment.

**Comment by mkoistinen** [comment]
Thank you!

**Pull request merged**